service/tbc/tbc_test.go

```diff
@@ -121,6 +122,61 @@ func countKeystones(b *btcutil.Block) int {
 	return keystonesFound
 }
 
+func TestDbUpgrade(t *testing.T) {
```

Comment:
if I add this change, which returns early from the "upgrade db for loop", this test still passes. therefore I do not believe it's actually testing the upgrade.  could we be more specific here? what _should_ the state of the db be before and after?
```
diff --git a/database/tbcd/level/level.go b/database/tbcd/level/level.go
index e1cefad..2375b5a 100644
--- a/database/tbcd/level/level.go
+++ b/database/tbcd/level/level.go
@@ -211,6 +211,7 @@ func New(ctx context.Context, cfg *Config) (*ldb, error) {
 
        // Upgrade database
        for {
+               return l, nil
                dbVersion, err := l.Version(ctx)
                if err != nil {
                        if errors.Is(err, leveldb.ErrNotFound) {
```

Reply:
I added a test to verify all end conditions of the upgrade and we added a new tar file that is moved from v1 to v2.